login + signup

trinity-y opened the pull request:
login signup works (for me at least)!!!! 

### things added that will affect everyone:
- added create() custom model to insert a row into your table
    - only inserts one row at a time
    - gets table metadata from postgres to get the col names and their types, cause strings and other datatypes require quotes around them. but i might not have caught all the types that need quotes (listed in `quotedTypes`) so beware
- modified findMany() custom model function to accept a WHERE clause similar to prisma, but rn it just ANDs everything 
- `base.js` returns the axios thing, see `auth.js` for how it's used
-  made a `MessageCard` component in 30 seconds thats EXTREMELY UGLY!!! so we should fix this later
- i removed `<script src="../src/index.js .... >` from `index.html` cause it was giving me errors... lmmk if that's bad

### overview of how login + signup works
**signup**
/signup, form sends link -> link goes to /complete-signup, logs user in and then sends post request to backend/users/ to fill in the db
**login**
/login, form sends link -> link redirects back to /login (idk if that's weird)
/login will always check if the current link is a signInLink (it knows because signInLinks come w/ query parameters in the url) and then it will try sign in.

### things that need to be fixed 
(may or may not do some of these by M1.... probably not though not gonna lie...)

- [ ] MessageCard is UGLY!!!!!
- [X] error messages
- [ ] should probably handle the situation where email is not found in localStorage
- [X] username validation on the frontend (it shouldnt add to the db if the username isnt unique but it doesn't tell the user properly)
- [ ] successful login redirects to '/' we don't have a homepage rn
- [ ] also idk if i can even deploy to GCP properly cause my docker is still broken? cause part of the process involves building the container? so idk

RohanNankani reviewed (approved):
Just one nit for later down the road: should have a feedback when signing up "Email sent" to make user friendly. 

Other than that, made sure sign up works and checked the DB.

LGTM! 🍻🚀

trinity-y commented:
**changes made**
- added username and email validation, 
- added email sent message, 
- modified findMany() to accept WHERE clause, 
- refactored custom a bit, 
- committed my startup script in case anyone wants to use it
- added frog coffee emojis 🐸☕🐸☕🐸☕
